test-utils/src/main/scala/com/mesosphere/utils/mesos/MesosTest.scala

```diff
@@ -0,0 +1,394 @@
+    // First-come-first-served task will bind successfully where the others will fail leading to a lot inconsistency and
+    // flakiness in tests.
+
+    val (faultDomainAgentAttributes: Map[String, Option[String]], mesosFaultDomainAgentCmdOption) = if (agentsFaultDomains.nonEmpty && agentsFaultDomains(i).nonEmpty) {
+      val faultDomain = agentsFaultDomains(i).get
+      val mesosFaultDomainCmdOption = s"""
+                                         |{
+                                         |  "fault_domain":
+                                         |    {
+                                         |      "region":
+                                         |        {
+                                         |          "name": "${faultDomain.region}"
+                                         |        },
+                                         |      "zone":
+                                         |        {
+                                         |          "name": "${faultDomain.zone}"
+                                         |        }
+                                         |    }
+                                         |}
+        """.stripMargin
+
+      val nodeAttributes = Map(
+        "fault_domain_region" -> Some(faultDomain.region.value),
+        "fault_domain_zone" -> Some(faultDomain.zone.value))
+      (nodeAttributes, Some(mesosFaultDomainCmdOption))
+    } else Map.empty -> None
+
+    // uniquely identify each agent node, useful for constraint matching
+    val attributes: Map[String, Option[String]] = Map("node" -> Some(i.toString)) ++ faultDomainAgentAttributes
+
+    val renderedAttributes: String = attributes.map { case (key, maybeVal) => s"$key${maybeVal.map(v => s":$v").getOrElse("")}" }.mkString(";")
+
+    Agent(resources = new Resources(ports = PortAllocator.portsRange(), gpus = agentsGpus), extraArgs = Seq(
+      s"--attributes=$renderedAttributes"
+    ) ++ mesosFaultDomainAgentCmdOption.map(fd => s"--domain=$fd"))
+  }
+
+  if (autoStart) {
+    start()
+  }
+
+  def start(): Future[String] = async {
+    masters.foreach(_.start())
+    agents.foreach(_.start())
+    val masterUri = await(waitForLeader())
+    await(waitForAgents(masterUri))
+    masterUri
+  }
+
+  def waitForLeader(): Future[String] = async {
+    val firstMaster = s"http://${masters.head.ip}:${masters.head.port}"
+    val result = Retry("wait for leader", maxAttempts = Int.MaxValue, maxDuration = waitForMesosTimeout) {
+      Http().singleRequest(Get(firstMaster + "/redirect")).map { result =>
+        result.discardEntityBytes() // forget about the body
+        if (result.status.isFailure()) {
+          throw new Exception(s"Couldn't determine leader: $result")
+        }
+        result
+      }
+    }
+
+    def maybeFixURI(uri: String): String = {
+      // some versions of mesos issue redirects with broken Location headers; fix them here
```

Comment:
Is this still the case?

Reply:
Who knows. Let's remove and see what happens